base/entries.go

```diff
@@ -0,0 +1,63 @@
+package base
+
+import (
+	"fmt"
+
+	"github.com/qri-io/dataset/dsio"
+)
+
+// ReadEntries reads entries and returns them as a native go array or map
+func ReadEntries(reader dsio.EntryReader, all bool, limit int, offset int) (interface{}, error) {
+	obj := make(map[string]interface{})
+	array := make([]interface{}, 0)
+	numRead := 0
+
+	tlt, err := dsio.GetTopLevelType(reader.Structure())
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0;; i++ {
+		val, err := reader.ReadEntry()
```

Comment:
think we can get away with using a `dsio.PagedReader` here?

Reply:
Even better, I think I can just move the PagedReader to base/render, the only place that needs to pass (all, limit, offset) to ReadEntries. It ends up making ReadEntries much simpler, though the test cases now become extremely trivial.